src/connect/render.cpp

```diff
@@ -348,7 +348,9 @@ namespace {
                     JSON_FIELD_INT_G(transfer_status.has_value(), "size", transfer_status->expected) JSON_COMMA;
                     JSON_FIELD_INT_G(transfer_status.has_value(), "transferred", transfer_status->transferred) JSON_COMMA;
                     JSON_FIELD_INT_G(transfer_status.has_value(), "time_transferring", (transfer_status->start - ticks_ms()) / 1000) JSON_COMMA;
-                    JSON_FIELD_STR_G(transfer_status.has_value(), "path", transfer_status->destination) JSON_COMMA;
+                    if (transfer_status->destination) {
+                        JSON_FIELD_STR_G(transfer_status.has_value(), "path", transfer_status->destination) JSON_COMMA;
+                    }
```

Comment:
I agree this'll actually work in practice, but is a bit fragile and non-trivial to prove (and is based on the fact destination never changes from non-null to null).

I think the reason _why_ this works needs a comment.